### PR TITLE
Add controllers, migrations,models,repositories and services

### PR DESCRIPTION
**Pull request by HerbertIV**

### CODE REVIEW

File: database/migrations/2022_02_02_133124_create_consultations_table.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('consultations', function (Blueprint $table) {
+            $table->id();
+            $table->bigInteger('author_id')->nullable()->index();
+            $table->bigInteger('media_id')->nullable()->index();
```
Comment (qunabu) [inline database/migrations/2022_02_02_133124_create_consultations_table.php:20]: what is `media_id` ?

File: database/migrations/2022_02_02_133124_create_consultations_table.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            $table->integer('base_price')->nullable();
+            $table->string('name', 80);
```
Comment (qunabu) [inline database/migrations/2022_02_02_133124_create_consultations_table.php:22]:
```suggestion
            $table->string('name', 255);
```

File: database/migrations/2022_02_02_135907_create_consultations_participants_table.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('consultations_participants', function (Blueprint $table) {
+            $table->id();
+            $table->bigInteger('consultation_id')->index();
+            $table->bigInteger('user_id')->index();
+            $table->string('status');
```
Comment (qunabu) [inline database/migrations/2022_02_02_135907_create_consultations_participants_table.php:20]: what is `status` ?

**Review by qunabu — CHANGES_REQUESTED** [review]

File: database/migrations/2022_02_02_135907_create_consultations_participants_table.php (hunk 3 above)
Comment (HerbertIV) [reply database/migrations/2022_02_02_135907_create_consultations_participants_table.php:20]: status determines whether the participant is connected or disconnected

File: database/factories/ConsultationParticipantFactory.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+use EscolaLms\Courses\Database\Factories\FakerMarkdownProvider\FakerProvider;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class ConsultationParticipantFactory extends Factory
+{
+    protected $model = ConsultationParticipant::class;
+
+    public function definition()
+    {
+        $this->faker->addProvider(new FakerProvider($this->faker));
```
Comment (pa-cholek) [inline database/factories/ConsultationParticipantFactory.php:18]: You are adding Markdown Faker provider and then not using it

File: database/migrations/2022_02_02_135907_create_consultations_participants_table.php (hunk 3 above)
Comment (pa-cholek) [reply database/migrations/2022_02_02_135907_create_consultations_participants_table.php:20]: Maybe add a default value to it `->default(Enum::VALUE)`

File: src/Http/Controllers/AppBaseController.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+use EscolaLms\Core\Http\Controllers\EscolaLmsBaseController;
+
+/**
+ * SWAGGER_VERSION
+ * This class should be parent class for other API controllers
+ * Class AppBaseController
+ */
+class AppBaseController extends EscolaLmsBaseController
+{
+    public function sendDataError($error, $data, $code = 422)
```
Comment (pa-cholek) [inline src/Http/Controllers/AppBaseController.php:14]: Are you using this function anywhere? If not, then your controllers should extend EscolaLmsBaseController directly

**Review by pa-cholek — APPROVED** [review]

File: src/Http/Controllers/AppBaseController.php (hunk 5 above)
Comment (HerbertIV) [reply src/Http/Controllers/AppBaseController.php]: Removed this controller

File: database/factories/ConsultationParticipantFactory.php (hunk 4 above)
Comment (HerbertIV) [reply database/factories/ConsultationParticipantFactory.php]: Removed this line

File: database/migrations/2022_02_02_135907_create_consultations_participants_table.php (hunk 3 above)
Comment (HerbertIV) [reply database/migrations/2022_02_02_135907_create_consultations_participants_table.php]: Added enum to participant status. Default status is disconnected

**Review by qunabu — APPROVED** [review]

**Pull request merged**